### PR TITLE
Add PRFinder to RoslynTools

### PR DESCRIPTION
**Pull request by RikkiGibson**
/cc @dibarbet @JoeRobich 

this version of it also will fetch and retry if it couldn't find the specified commits (likely, when the commit being inserted was a very recent merge commit to master-vs-deps)

### CODE REVIEW

File: src/RoslynInsertionTool/PRFinder/Program.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                }
+            }
+
+            return 0;
+        }
+
+        private static void CurrentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs e)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (JoeRobich) [inline src/RoslynInsertionTool/PRFinder/Program.cs]: Is this used?

Comment (RikkiGibson) [reply src/RoslynInsertionTool/PRFinder/Program.cs]: seems not :)

**Comment by RikkiGibson** [comment]
So the plan here is to make PRFinder unnecessary once we get the chance, by fixing RIT to show the changesets properly. I'm going to make some time this afternoon to try and make that happen.

**Comment by dibarbet** [comment]
> So the plan here is to make PRFinder unnecessary once we get the chance, by fixing RIT to show the changesets properly

I'm not sure what the plan is exactly for vs mac insertions, but we'll probably need the same tool when doing those

**Pull request merged**